### PR TITLE
[#5864] Learning Blue Magic 

### PR DESCRIPTION
**Pull request by zach2good**
Original Issue:
https://github.com/DarkstarProject/darkstar/issues/5864

Using this list:
http://ffxi.allakhazam.com/db/skillmods.html?fskillmod=1321

And this:
https://ffxiclopedia.fandom.com/wiki/Category:Blue_Magic#How_to_Learn_Blue_Magic
`The Magus Bazubands/Magus Bazubands +1/Assimilator's Bazubands/Assimilator's Bazubands +1 are the only known way to directly increase the chance to learn Blue Magic.`

And all entries (real or not) of Assimilator's Bazubands in the items db

Tested by logging out `chanceToLearn ` while getting hit by 1000 needles and bomb toss as a blu with and without the AF gloves

### CODE REVIEW

**Comment by Hozu** [comment]
You'll have to create a modifier for the bonus, and assign it to the appropriate equips. Not only does this mean that the items won't be hardcoded in the core (ie. no need to modify that part if new equipment comes out with those effects), but modifiers are affected by level sync (ie. this wouldn't work below equip level), while equip ID checks are not.

File: src/map/utils/blueutils.cpp [hunk 1]
```diff
@@ -136,8 +136,18 @@ void TryLearningSpells(CCharEntity* PChar, CMobEntity* PMob) {
             // make sure the difference between spell skill and player is at most 31 points
             if (playerSkillLvl >= skillLvlForSpell - 31)
             {
-                // TODO: check for blue learning bonus and adjust base percent
-                if (dsprand::GetRandomNumber(100) < 33) {
+                uint chanceToLearn = 33;
```
Comment (nfaletra) [inline src/map/utils/blueutils.cpp]: This needs to be a mod. Do not hardcode item ids like this.

**Comment by zach2good** [comment]
OK, I'll look into it 👍

**Comment by zach2good** [comment]
Thanks for all the help @nfaletra, all the points should be covered now

**Comment by nfaletra** [comment]
No problem. As a side note, I personally think `auto` for simple types is bad. Not everyone on the project agrees with that so it's fine here. I tend to only use it for complex types (like iterators and lambdas). I feel it promotes laziness and hinders readability for someone else looking at the code when types are explicitly stated.

**Pull request merged**